lib/product-import.coffee

```diff
@@ -276,11 +277,19 @@ class ProductImport
         productToProcess.categories = _.map prodCatsIds, (catId) ->
           id: catId
           typeId: 'category'
-      if not productToProcess.slug
-        debug 'slug missing in product to process, assigning same as existing product: %s', existingProduct.slug
-        productToProcess.slug = existingProduct.slug # to prevent removing slug from existing product.
+      productToProcess.slug = @_updateProductSlug productToProcess, existingProduct
       Promise.resolve productToProcess
 
+  _updateProductSlug: (productToProcess, existingProduct) =>
+    if @ignoreSlugUpdates
+      slug = existingProduct.slug
+    else if not productToProcess.slug
```

Comment:
I wanted to give a specific debug message in the case of `not productToProcess.slug` that is why I kept it separate.